elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -424,6 +434,14 @@ export class PeerPool extends EventEmitter {
 		socket: SCServerSocket,
 	): Peer {
 		const inboundPeers = this.getPeers(InboundPeer);
+		const peerId = constructPeerIdFromPeerInfo(peerInfo);
+
+		if (this._peerMapDuplicates.has(peerId) && this._peerMap.has(peerId)) {
+			throw new Error(
+				`Peer ${peerId} already has inbound and outbound connections.`,
+			);
```

Comment:
This vulnerability was also present in the old version, if we try to add already existing peer in peerMap it will crash. However, it was checked in p2p.js before we call `addIbound` thats why it was not happening. But yes, it will crash the node with this error being thrown so now I will not throw error but will simply emit event and return.